obsreport/obsreporttest/obsreporttest.go

```diff
@@ -44,122 +45,152 @@ var (
 
 // SetupRecordedMetricsTest does setup the testing environment to check the metrics recorded by receivers, producers or exporters.
 // The returned function should be deferred.
-func SetupRecordedMetricsTest() (func(), error) {
+func SetupRecordedMetricsTest() (component.TelemetrySettings, func(), error) {
```

Comment:
I think you need to return more than just `component.TelemetrySettings`, for otel we will need to also return some "inmemory"/"test" exporters.

So probably need to return:
```golang
type BlaBla struct {
TelemetrySettings component.TelemetrySettings
SpanRecorder *tracetest.SpanRecorder
... // will add something for logs (observer I think is called)
... // will add something for metrics
}
```

Reply:
Probably should be also called "SetupTestTelemetrySettings" :)

Reply:
BlaBla -> TestTelemetrySettings

Reply:
`BlaBla` would be easy to search for and type though 🤔